app/consumer.py

```diff
@@ -26,57 +26,92 @@
 from app.helpers.exceptions import BadMessageError, DecryptError, RetryableError
 
 
-def get_delivery_count_from_properties(properties):
-    """
-    Returns the delivery count for a message from the rabbit queue. The
-    value is auto-set by rabbitmq.
-    """
-    delivery_count = 0
-    if properties.headers and 'x-delivery-count' in properties.headers:
-        delivery_count = properties.headers['x-delivery-count']
-    return delivery_count + 1
-
-
 class Consumer(AsyncConsumer):
 
     def __init__(self, args=None, cfg=None):
         self._args = args
         self._cfg = cfg
-        self.quarantine_publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_QUARANTINE_QUEUE)
+        self.quarantine_publisher = QueuePublisher(logger,
+                                                   settings.RABBIT_URLS,
+                                                   settings.RABBIT_QUARANTINE_QUEUE)
         super().__init__()
 
+    def get_delivery_count_from_properties(self, properties):
+        """
+        Returns the delivery count for a message from the rabbit queue. The
+        value is auto-set by rabbitmq.
+        """
+        delivery_count = 0
+        if properties.headers and 'x-delivery-count' in properties.headers:
+            delivery_count = properties.headers['x-delivery-count']
+        return delivery_count + 1
+
+    def get_tx_id_from_properties(self, properties):
+        """
+        Returns the tx_id for a message from a rabbit queue. The value is
+        auto-set by rabbitmq.
+        """
+        try:
+            tx_id = properties.headers['tx_id']
+            logger.info("Retrieved tx_id from message properties", tx_id=tx_id)
+            return tx_id
+        except KeyError as e:
+            logger.error("No tx_id in message properties. Sending message to quarantine")
+            raise e
+
```

Comment:
Neither _get_delivery_count_from_properties_ or _get_tx_id_from_properties_ use the _self_ object. This means they could become static methods, hence callable elsewhere without needing a Consumer object to be instantiated.